imagetool/src/main/java/com/oracle/weblogic/imagetool/util/Utils.java

```diff
@@ -632,4 +632,30 @@ public static String getOpatchVersionFromZip(String fileName) {
         return null;
     }
 
+    /**
+     * validatePatchIds validate the format of the patch ids
+     * @param patches list of patch ids
+     * @throws IllegalAccessException when the first id failed the format validation
+     */
+
+    public static void validatePatchIds(List<String> patches) throws IllegalAccessException {
+        Pattern patchIdPattern = Pattern.compile(Constants.PATCH_ID_REGEX);
+        if (patches != null && !patches.isEmpty()) {
+            for (String patchId : patches) {
+                Matcher matcher = patchIdPattern.matcher(patchId);
+                if (!matcher.matches()) {
+                    String error = String.format("Invalid patch id %s. Patch id must be in the format of 12345678[_12"
+                        + ".2.1.3.0]:  "
+                        + "started with 8 digits patch id.  If the patch has different version, "
```

Comment:
*starting with 8 digit patch ID. For patches that have multiple target versions, the target version must be specified after the underscore with 5 places, such as 12.2.1.3.0 or 12.2.1.3.190416

Reply:
I tried multiple patches. It seems that the command 'imagetool cache addPatch' requires patch version suffix such as '_12.2.1.3.0' even there is only one target version in support.us.oracle.com